src/components/Movies/MovieFilterBar.tsx

```diff
@@ -1,58 +1,120 @@
-import React, { FC, MouseEventHandler } from 'react'
-import classNames from "classnames"
+import React, { FC } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { changeSortOrder, setGenre, setSortBy } from '../../features/movies/moviesSlice'
+import { RootState } from '../../app/store'
+import classNames from 'classnames'
 
 import styles from './MovieFilterBar.scss'
+import useToggle from '../../utils/useToggle'
 
-import {SortOrderType} from "../../App";
-
-const genres = ['All','Documentary','Comedy','Horror','Crime']
+export interface GenreType {
+  id: string
+  title: string
+}
 
-interface MovieFilterBarProps {
-  sortOrder: SortOrderType
-  onSortOrderChange: (k: SortOrderType) => void
+export interface SortType {
+  id: string
+  title: string
 }
 
-const MovieFilterBar: FC<MovieFilterBarProps> = ({
-  sortOrder,
-  onSortOrderChange
-}) => {
-  const activeGenre = 'All'
-  const activeSort = 'Release Date'
+const sortTypes: SortType[] = [
+  {
+    id: 'release_date',
+    title: 'Release Date'
+  },
+  {
+    id: 'genre',
+    title: 'Genre'
+  },
+  {
+    id: 'rating',
+    title: 'Rating'
+  },
+]
+
+const genres: GenreType[] = [
+  {
+    id: 'all',
+    title: 'All'
+  },
+  {
+    id: 'documentary',
+    title: 'Documentary'
+  },
+  {
+    id: 'comedy',
+    title: 'Comedy'
+  },
+  {
+    id: 'drama',
+    title: 'Drama'
+  },
+  {
+    id: 'romance',
+    title: 'Romance'
+  },
+]
+
+const MovieFilterBar: FC = () => {
+  const dispatch = useDispatch()
+  const [areSortOrdersVisible, toggleSortOrderVisibility] = useToggle(false)
+  const sortOrder = useSelector((state: RootState) => state.movies.sortOrder)
+  const sortByTitle = useSelector((state: RootState) => state.movies.sortByTitle)
+  const genreSelected = useSelector((state: RootState) => state.movies.genreSelected)
 
   const invertSortOrder = () => {
-    if (sortOrder === 'asc') {
-      onSortOrderChange('desc')
-    } else {
-      onSortOrderChange('asc')
-    }
+    dispatch(changeSortOrder())
+  }
+
+  const changeGenreType = (genre: GenreType) => {
+    dispatch(setGenre(genre))
+  }
+
+  const changeSortType = (sortType: SortType) => {
+    dispatch(setSortBy(sortType))
+    toggleSortOrderVisibility()
   }
 
   return (
     <div className={styles['movieFilterBar']}>
       <ul className={styles['movieFilterBar--genres']}>
         {genres.map(genre => {
           return (<li
+            onClick={() => changeGenreType(genre)}
```

Comment:
Arrow functions as a prop are an anti-pattern. It's critical though but if you have time please refactor this.